src/Kiota.Builder/Refiners/CodeUsingComparer.cs

```diff
@@ -14,12 +14,14 @@ public CodeUsingComparer(bool compareOnDeclaration)
     }
     public bool Equals(CodeUsing? x, CodeUsing? y)
     {
-        return (!_compareOnDeclaration || x?.Declaration == y?.Declaration) && (x?.Name?.Equals(y?.Name, StringComparison.Ordinal) ?? false);
+        return (!_compareOnDeclaration || x?.Declaration == y?.Declaration) && string.Equals(x?.Name, y?.Name, StringComparison.Ordinal);
```

Comment:
same comment about using the same comparer instance for both the equals and GetHashCode implementations.
Not saying we need to make it customizable via an additional constructor.